### ISSUE TITLE
Unable to close Tailscale menu on Chromecast with Google TV

### ISSUE THREAD

**Issue by garrett-ts** [tailscale/tailscale#7211]
### What is the issue?

Reported from a support ticket ([9242](https://secure.helpscout.net/conversation/2139727546/9242?folderId=6988689))

When using a Chromecast with Google TV device, opening the Tailscale menu results in not being able to close the menu. The only available option is to force quit the app then re-open.

### OS

Android

### OS version

12

### Tailscale version

1.36.0

**Comment by MDSharma** [comment]
Exact same issue here.. looking forward to finding a solution

**Comment by tinderboxmedia** [comment]
Same issue! It's even worse the moment you open the QR code that links to this GitHub. It will block the entire screen and you won't be able to do anything about it.

**Comment by DentonGentry** [comment]
I can reproduce this locally. It broke when we updated gioui.org in 1.36 with https://github.com/tailscale/tailscale-android/pull/73

The problem with the Back key appears similar to https://todo.sr.ht/~eliasnaur/gio/408

If I add a call to `key.InputOp{Tag: d, Keys: key.NameBack}.Add(gtx.Ops)` I can get the back button delivered, but the up and down events from the remote control stop working. So the menu can display but you cannot advance to any of the menu items.

If I make the call be:
```
+       keys := key.Set(key.NameBack + "|" + key.NameUpArrow + "|" + key.NameDownArrow)
+       key.InputOp{Tag: d, Keys: keys}.Add(gtx.Ops)
```
I can get the Up and Down events from the remote to be delivered to the app, but it never had to deal with its own menu selection before (because Gio was doing it) and it has no code to do so. Still trying to figure out if we can either:
1. Only receive Back and let Gio receive Up and Down
1. Pass the Up and Down events back to Gio to deal with.

**Comment by z3ntu** [comment]
Is there any workaround like an older .apk available without this bug? I think it's impossible for me to log in since accessing the menu for putting in a custom server only appears after going clicking on an item multiple times but I cannot close the overflow menu again. And for anything else force closing the app from the settings isn't great either.

**Comment by z3ntu** [comment]
I managed to work around this problem by connecting a Bluetooth mouse to the "Chromecast with Google TV" and using the mouse to click (and close) the overflow menu to get to the server url input.

**Comment by soleous** [comment]
I'm also getting hit by this.  Looking at the appstore comments the app is getting bad reviews because of this.  Any fix planned, it also looks like it was brought up specically regarding Nvidia shield in 2022, but no fix was planned?

**Comment by wlallemand** [comment]
Any update about this? This is simply unusable.

**Comment by danny50610** [comment]
Not sure if it helps

But after several hours of trying, including comments provided by DentonGentry (https://github.com/tailscale/tailscale/issues/7211#issuecomment-1465083103)

Although the up and down keys cannot be used, the left and right keys can.

Maybe add a back button directly to the UI as a compromise solution  😅

```diff
diff --git a/cmd/tailscale/main.go b/cmd/tailscale/main.go
index 5360574..bf0d6d6 100644
--- a/cmd/tailscale/main.go
+++ b/cmd/tailscale/main.go
@@ -1202,6 +1202,19 @@ func (a *App) runUI() error {
 				e.Insets = system.Insets{}
 				gtx := layout.NewContext(&ops, e)
 				events := ui.layout(gtx, ins, state)
+
+				for _, event := range gtx.Events(w) {
+					switch e := event.(type) {
+					case key.Event:
+						if e.Name == key.NameBack {
+							if ui.onBack() {
+								w.Invalidate()
+							}
+						}
+					}
+				}
+				key.InputOp{Tag: w, Keys: key.Set(key.NameBack)}.Add(gtx.Ops)
+
 				e.Frame(gtx.Ops)
 				a.processUIEvents(w, events, activity, state, files)
 			}

```


https://github.com/tailscale/tailscale/assets/3949245/be2fdc86-152e-4437-af02-115c26a2d7a3

**Comment by nickkhyl** [comment]
Hi,
I can assure you that this issue is on our roadmap and will be addressed.
Unfortunately, we must tackle some technical debt first, and our Android team is actively working on it.

**Comment by soccerties** [comment]
I was able to work around this by clicking "copy ip address". That will close the menu and you can access the main screen again.

**Comment by agottardo** [comment]
This should be fixed in the new Android app (v1.66 or later).
